drivers/spi/spi_nrfx_spim.c

```diff
@@ -43,19 +43,20 @@ LOG_MODULE_REGISTER(spi_nrfx_spim, CONFIG_SPI_LOG_LEVEL);
 #define SPI_BUFFER_IN_RAM 1
 #endif
 
-#if defined(CONFIG_CLOCK_CONTROL_NRF2_GLOBAL_HSFLL) && \
-	(defined(CONFIG_HAS_HW_NRF_SPIM120) || \
-	 defined(CONFIG_HAS_HW_NRF_SPIM121))
-#define SPIM_REQUESTS_CLOCK(idx) UTIL_OR(IS_EQ(idx, 120), \
-					 IS_EQ(idx, 121))
-#define USE_CLOCK_REQUESTS 1
+#if defined(CONFIG_CLOCK_CONTROL_NRF2_GLOBAL_HSFLL)
+#define SPIM_REQUESTS_CLOCK(idx) DT_NODE_HAS_COMPAT(DT_NODELABEL(DT_CLOCKS_CTLR(idx)), \
+				 nordic_nrf_hsfll_global)
+#define SPIM_REQUESTS_CLOCK_OR(idx) SPIM_REQUESTS_CLOCK(idx) ||
+#define SPIM_REQUESTS_CLOCK_ANY (DT_FOREACH_STATUS_OKAY(nordic_nrf_spim, SPIM_REQUESTS_CLOCK_OR) 0)
+#define USE_CLOCK_REQUESTS SPIM_REQUESTS_CLOCK_ANY
```

Comment:
nit: this will give much more readable output in case any error in processing of this `FOREACH` macro should happen, as it will be processed only in one place (and not inside `SPI_NRFX_SPIM_DEFINE()`, what is the worst thing)
```suggestion
#if (DT_FOREACH_STATUS_OKAY(nordic_nrf_spim, SPIM_REQUESTS_CLOCK_OR) 0)
#define USE_CLOCK_REQUESTS 1
#endif
```